src/libtiled/tilelayer.h

```diff
@@ -568,6 +571,16 @@ inline void TileLayer::setSize(const QSize &size)
     mHeight = size.height();
 }
 
+inline QRect TileLayer::bounds() const
+{
+    int x = std::min(0, mBounds.x());
+    int y = std::min(0, mBounds.y());
+    int width = std::max(mWidth, mBounds.width());
+    int height = std::max(mHeight, mBounds.height());
```

Comment:
```
QRegion TilePainter::paintableRegion(const QRegion &region) const
{
    const QRegion bounds = QRegion(mTileLayer->bounds());
    QRegion intersection = bounds.intersected(region);

    const QRegion &selection = mMapDocument->selectedArea();
    if (!selection.isEmpty())
        intersection &= selection;

    return intersection;
}
```

Due to this, if we directly return `mBounds` then we won't be able to paint the tile layer.